src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.Replace.cs

```diff
@@ -171,7 +173,9 @@ private static string Replace(MatchEvaluator evaluator, Regex regex, string inpu
             }
             else
             {
-                StringBuilder sb = StringBuilderCache.Acquire();
+                ReadOnlySpan<char> inputSpan = input.AsSpan();
```

Comment:
Is there a benefit to doing this `inputSpan` thing rather than at all of the `inputSpan.Slice(...)` call sites change them to be `input.AsSpan(...)`?

Reply:
It saves us creating a Span of the input string for every character to encode inside the loop.

Reply:
ok now I understood what you meant, you are right, there's no need for the inputSpan local.